{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://www.consevol.org/pdf/Navas-Molina_2013_MethodsEnzymol.pdf  \n",
    "http://cafe.daum.net/_c21_/bbs_search_read?grpid=n91&fldid=3G4&datanum=102   \n",
    "[Jackkiniffeì— ê´€í•œ ë…¼ë¬¸](https://www.researchgate.net/profile/Ruofei_Du/publication/323595926_A_weighted_Jackknife_method_for_clustered_data/links/5b3b9e174585150d23f64029/A-weighted-Jackknife-method-for-clustered-data.pdf?origin=publication_detail)  \n",
    "í•˜ê¸° ë‚´ìš©ì€ ìƒê¸° ë§í¬ë¥¼ ì •ë¦¬í•œê²ƒ  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Advancing Our Understanding of the Human Microbiome Using Qiime"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Abstract\n",
    "DNA Sequencingì˜ ë†’ì€ ì²˜ë¦¬ëŸ‰ ê¸°ìˆ ì€ bioinfomatics(ìƒë¬¼ ì •ë³´í•™) toolsì™€ ë°œì „ê³¼ í•¨ê»˜ í•´ì™”ê³ ,  \n",
    "microbial ecology(ë¯¸ìƒë¬¼ ìƒíƒœí•™)ì´ ê¸‰ê²©í•œ ë°œì „ì„ ì´ë£¨ê²Œ í•˜ì˜€ìœ¼ë©°, ì‚¬ëŒì˜ bioinfomaticsë¥¼ ì´í•´í•˜ê²Œ ë˜ëŠ”ë° ê¸°ì—¬í•˜ì˜€ë‹¤.  \n",
    "ë§ì€ toolsì¤‘ Qiime(Quantiative Insights Into Microbial Ecology)ê°€ ìˆë‹¤.\n",
    "> * DNA sequence dataì— ëŒ€í•œ mirobial community ë¶„ì„í•˜ê¸° ìœ„í•´ ë§Œë“¤ì–´ì§„ opensource bioinformatics ì†Œí”„íŠ¸ì›¨ì–´ íŒ¨í‚¤ì§€ì´ë‹¤.  \n",
    "* í†µê³„ ë¶„ì„ì„ í†µí•˜ì—¬ ì¶œë ¥ì´ ë˜ê³  ì¶œë ¥ê²°ê³¼ë¥¼ Visualizationí•  ìˆ˜ ìˆë‹¤.  \n",
    "  \n",
    "ì´ ë²ˆ ì±•í„°ì—ì„œëŠ”, ì‹ ì²´ ëª‡ êµ°ë°ì—ì„œ ì¶”ì¶œí•œ transgenicì„ QIIME mirobial communities íŒŒì´í”„ ë¼ì¸ì„ í†µí•´ ë§Œë“¤ì–´ì§„ ë°ì´í„°ì™€,  \n",
    "illumina MiSeq platformì„ í†µí•´ ë§Œë“¤ì–´ì§„ ì•¼ìƒ ì¥ì˜ 16S rRNA ìœ ì „ì sequencesë¥¼ ì‹œì—°í•´ë³¼ê²ƒì´ë‹¤.  \n",
    "í˜„ì¬ ì‹¤í–‰ ê°€ëŠ¥í•œ íŒŒì´í”„ë¼ì¸ ì¶”ì²œí•˜ìë©´, microbial community analysisì™€ ê°€ì´ë“œ ë¼ì¸ì—ì„œ ì œê³µí•˜ëŠ” í”„ë¡œì„¸ìŠ¤ì˜ critical choicesì´ë‹¤.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. INTRODUCTION\n",
    "DNA ì‹œí€€ì‹± ê¸°ìˆ ì˜ ë°œì „ì€ ëŒ€ëŸ‰ì˜ ë°ì´í„°ë¥¼ ë¶„ì„í•˜ê¸° ìœ„í•œ culture-independent ì‹œí€€ì‹±ê³¼ ì†Œí”„íŠ¸ì›¨ì–´ì˜ ê°€ìš©ì„±ê³¼ í•¨ê»˜ ë‹¤ì–‘í•œ í™˜ê²½ì—ì„œ ë¯¸ìƒë¬¼ êµ°ì§‘ì„ íŠ¹ì„±í™” í•˜ëŠ”ë° ëŠ¥ë ¥ì„ í¬ê²Œ í–¥ìƒì‹œì¼°ë‹¤.  \n",
    "ì¸ê°„ì˜ ë¯¸ìƒë¬¼(ì¸ì²´ì— ì¡´ì¬í•˜ëŠ” ë¯¸ìƒë¬¼ ì§‘í•©ì²´)ì€ ê´€ì‹¬ì˜ ëŒ€ìƒì´ë©°, ë¯¸ìƒë¬¼ ì„¸í¬ëŠ” ìš°ë¦¬ ëª¸ì˜ ì„¸í¬ë³´ë‹¤ 10:1ì˜ ë¹„ìœ¨ë¡œ ë” ë§ë‹¤.\n",
    "ë˜í•œ ì¸ê°„ì˜ ë¯¸ìƒë¬¼ êµ°ì§‘ì€ ê±´ê°•í•œ human microbiotaì— ê¸°ì—¬í•˜ê³  ìˆë‹¤.  \n",
    "ë¯¸ìƒë¬¼ì€ development, dysbiosis(or imbalance in these communities), ë¹„ë§Œ, í¬ë¡ ë³‘ì„ ë¹„ë¡¯í•œ ì§ˆë³‘ê³¼ë„ ê´€ê³„ê°€ ê¹Šë‹¤. \n",
    "\n",
    "ë¬´ê· ìƒì¥ë¥¼ ì´ìš©í•œ ì‹¤í—˜ì„ í†µí•´ ë¯¸ìƒë¬¼ì´ ì›ì¸ê· ì´ ë ìˆ˜ ìˆìŒì„ ì‹œì‚¬í–ˆë‹¤. ì™œëƒí•˜ë©´ certain phenotypes ê°€ ë¯¸ìƒë¬¼ì„ ì ì—¼ì‹œì¼œ ì „ë‹¬ ë ìˆ˜ ìˆê¸°ë•Œë¬¸ì´ë‹¤. (Carvalho et al., 2012; McLean, Bergonzelli, Collins, & Bercik, 2012; (Diaz Heijtz et al., 2011; Koren et al., 2012, Smith et al., 2013).  \n",
    "  \n",
    "ìµœê·¼ ì—°êµ¬ì—ì„œ ê¸‰ê²©í•˜ê²Œ ì¦ê°€í•˜ê¸° ì‹œì‘í•œ ë¯¸ìƒë¬¼ ë°ì´í„°ë¥¼ ì²˜ë¦¬í•˜ê¸° ìœ„í•œ ë„êµ¬ì˜ í•„ìš”ì„±ì´ ëŒ€ë‘ë˜ì—ˆë‹¤.  \n",
    "QIIME(Quantitative Insights Into Microbial Ecology)ëŠ” raw sequence dataì˜ ìƒí˜¸ ì‘ìš©ìœ¼ë¡œ í†µê³„, ì‹œê°í™”, ë¶„ì„ì— ì´ë¥´ê¸°ê¹Œì§€ self-contained microbial community ë¶„ì„ì„ ì œê³µí•˜ë„ë¡ ê³ ì•ˆëœ ì˜¤í”ˆ ì†ŒìŠ¤ íŒŒì´í”„ ë¼ì¸ì´ë‹¤.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. QIIME AS INTEGRATED PIPELINE OF THIRD-PARTY TOOLS  \n",
    "ì´ì „ê¹Œì§€ QIIMEì˜ ê°€ì¥ í° ë¬¸ì œëŠ” ì„¤ì¹˜ì˜€ë‹¤.  \n",
    "ì™œëƒí•˜ë©´ QIIMEì€ ì—¬ëŸ¬ê°€ì§€ ì†Œí”„íŠ¸ê°€ ë…ë¦½ì ì´ì˜€ê¸° ë•Œë¬¸ì´ë‹¤.\n",
    "ê·¸ëŸ¬ë‚˜ ì§€ì†ì ì¸ ì—…ë°ì´íŠ¸ë¥¼ í†µí•˜ì—¬ QIIMEìì²´ì˜ ë¶„ì„ì•Œê³ ë¦¬ì¦˜ì´ ë‚´ì¥ë˜ì—ˆë‹¤.  \n",
    "\n",
    "í•˜ì§€ë§Œ íŒŒì´í”„ ë¼ì¸ì—ëŠ” ë‘ê°€ì§€ ì˜µì…˜ì´ ì¡´ì¬í•œë‹¤.  \n",
    "ì•Œê³ ë¦¬ì¦˜ì„ ë‹¤ì‹œ êµ¬í˜„í•˜ê±°ë‚˜ ê¸°ì¡´ì— ë‚´ì¥ëœ íŒŒì´í”„ë¼ì¸ì„ ì‚¬ìš©í•˜ëŠ” ê²ƒì´ë‹¤.  \n",
    "\n",
    "QIIMEì€ ì•Œê³ ë¦¬ì¦˜ì„ ë‹¤ì‹œ êµ¬í˜„í•˜ëŠ” ëŒ€ì‹  ëª¨ë“  ì•Œê³ ë¦¬ì¦˜ì„ í¬í•¨í•˜ëŠ”ê²ƒìœ¼ë¡œ ê°œì¸ì˜ ë¬´ê²°ì„±ì„ ë³´ì¥í•œë‹¤.  \n",
    "ì‚¬ìš©í•˜ëŠ” ë„êµ¬ ìì²´ë¥¼ ì›ì €ìê°€ ì„¤ê³„, ì‘ì„± ë° í…ŒìŠ¤íŠ¸ í•œê²ƒì´ë©°, ëŒ€ë¶€ë¶„ì˜ ê²½ìš° ê³¼í•™ ê³µë™ì²´ë¡œ ë¶€í„° ì‹¬ì‚¬ë¥¼ ë°›ì•˜ë‹¤.  \n",
    "  \n",
    "ë˜í•œ ì§€ì†ì ìœ¼ë¡œ ì—…ë°ì´íŠ¸ ë˜ëŠ” ì•Œê³ ë¦¬ì¦˜ì„ ìƒˆë¡œ ì ìš©ì‹œì¼œ ì‹ ì†í•˜ê²Œ ë°°í¬í•˜ê³  ìˆë‹¤.\n",
    "\n",
    "QIIME Parameterì˜ ê¸°ë³¸ê°’ì€ **uclust**ì´ê³ , cluster sequencesëŠ” **The GreenGenes**ë¥¼ ê¸°ë°˜ìœ¼ë¡œ í•˜ì˜€ìœ¼ë©°,  \n",
    "taxonomyì—ëŠ” **RDP classifier**ì„ ì‚¬ìš©í•˜ì˜€ë‹¤. \n",
    "ê·¸ë¦¬ê³  ê·¸ë§Œí¼ ì¤‘ìš”í•œ ë¶„ì„(PCoA)ì€ **UniFrac**ì´ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ê°ê°ì˜ íŒŒì´í”„ ë¼ì¸ì€ ì—°êµ¬ì›ë“¤ì˜ ì°¸ì—¬í•˜ê³  í™•ì¥í•˜ê³  ìˆë‹¤.  \n",
    "ì˜ˆë¥¼ ë“¤ì–´ pick_otus.pyëŠ” 9ê°€ì§€ì˜ ë‹¤ë¥¸ clustering ì•Œê³ ë¦¬ì¦˜ì„ ì œê³µí•˜ë©°, ê°ê°ì˜ ì•Œê³ ë¦¬ì¦˜ì€ ì¸ì •ë°›ëŠ” ì—°êµ¬ì›ì´ ì œì‘í•˜ì˜€ë‹¤.  \n",
    "  \n",
    "taxonomy ë°ì´í„° ë² ì´ë‚˜ reference dataesì—ëŠ” GreenGenesê°€ í¬í•¨ë˜ì–´ ìˆê³  ê·¸ê²ƒì´ release ë²„ì „ì— ì¤‘ìš”í•˜ê²Œ ì¸ìš©ë˜ì–´ ìˆë‹¤. ë”°ë¼ì„œ ê°™ì€ ê²°ê³¼ë¥¼ ë„ì¶œí•˜ê¸° ìœ„í•´ì„œëŠ” ë¦´ë¦¬ì¦ˆì— ì‚¬ìš©ë˜ì—ˆë˜ dataë¥¼ ì‚¬ìš©í•˜ëŠ” ê²ƒì´ ì¤‘ìš”í•˜ë‹¤.  \n",
    "  \n",
    "ì‚¬ìš©ëœ ë²„ì „ì€ release date ì´í›„ì— í‘œì‹œë˜ì–´ ìˆë‹¤.\n",
    "* 12_10 release.  \n",
    "  \n",
    "GreenGenesëŠ” giime.org ì›¹ì—ì„œ ë°›ì„ ìˆ˜ ìˆë‹¤.    \n",
    "  \n",
    "ì•„ë˜ëŠ” QIIME íŒŒì´í”„ ë¼ì¸ì— ì‚¬ìš©ëœ íˆ´ë“¤ì´ë‹¤.  \n",
    "> â€¢ uclust (Edgar, 2010). Used for OTU picking.  \n",
    "â€¢ usearch (Edgar, 2010). Used for OTU picking and chimera checking. 374 JosÃ© A. Navas-Molina et al. Author's personal copy  \n",
    "â€¢ RDP classifier (Wang et al., 2007). Used for taxonomy assignment.  \n",
    "â€¢ GreenGenes database (DeSantis etal., 2006). Used as a reference database for taxonomy assignment and reference-based OTU picking (see below).  \n",
    "â€¢ PyNAST (Caporaso, Bittinger, et al., 2010). Used for multiple sequence alignment.  \n",
    "â€¢ UniFrac (Lozupone & Knight, 2005). Used as a phylogenetic metric for beta-diversity analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. PCR AND SEQUENCING ON ILLUMINA MISeq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PCR ë‹¨ê³„ì—ì„œ sequencing adaptorsì™€ ë°”ì½”ë“œë¥¼ ë„ì…í•˜ë©´ í•„ìš”í•œ 16S amplicon ì˜ ì—¬ëŸ¬ íš¨ì†Œ ì²˜ë¦¬ë³´ë‹¤ ì´ì ì´ ìˆë‹¤.  \n",
    "\n",
    "ì²«ë²ˆì§¸ë¡œ ê³ ë ¤í•´ì•¼í•  ì¤‘ìš”ì‚¬í•­ì€ sample handlingì„ ì¤„ì´ëŠ” ê²ƒì´ë‹¤.  \n",
    "* sample handlingì„ ì¤„ì´ê²Œ ëœë‹¤ë©´ ìƒ˜í”Œì˜ ì˜¤ì—¼ ìœ„í—˜ì„±, ì˜ëª»ëœ ë¼ë²¨ë§, ê·¸ë¦¬ê³  ì¤€ë¹„ ê³¼ì •ì—ì„œ ìƒ˜í”Œì˜ ì†ì‹¤ì„ ìµœì†Œí™” í•  ìˆ˜ ìˆë‹¤.  \n",
    "  \n",
    "PCR ë‹¨ê³„ì—ì„œ adapterì™€ ë°”ì½”ë“œë¥¼ ê²°í•©í•˜ë©´ ìƒ˜í”Œì˜ primersê°€ ì •í™•í•˜ê²Œ ë§µí•‘í•˜ì—¬ ìƒ˜í”Œì˜ ë°”ì½”ë“œë¥¼ ì •í™•í•˜ê²Œ ì•Œ ìˆ˜ ìˆê²Œ ëœë‹¤.  \n",
    "  \n",
    "QIIME ê¸°ë³¸ ì„¤ì •ì€ Golay barcodeë¥¼ ê°ì§€í•˜ë¯€ë¡œ QIIMEì½”ë“œë¥¼ ì´ìš©í•˜ì—¬ ì‹œí€€ìŠ¤ë¥¼ ë³´ì¡´í•˜ê³  ì •í™•ë„ë¥¼ ê·¹ëŒ€í™” í• ìˆ˜ ìˆë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "custom barcodesë¡œ ì‹¤í–‰ë˜ëŠ” ampliconìš© MiSeq loadì— ëŒ€í•œ ìì„¸í•œ ë‚´ìš© ì•„ë˜ ì‚¬ì´íŠ¸ë¥¼ í™•ì¸í•´ë³´ë©´ëœë‹¤.\n",
    "(www.earthmicrobiome.org)\n",
    "\n",
    "ì¦‰, poolëœ librariesëŠ” Bioanalyzer(Agilent Technologies)ì— ì˜í•´ ë¶„ì„ë˜ê³  Qubit Fluorometer(Life Technologies, high-sensitivity reagents)ë¥¼ ì´ìš©í•˜ì—¬ 2$\\eta$Mìœ¼ë¡œ í¬ì„ëœë‹¤.  \n",
    "  \n",
    "pHiX spike-in library(illumina Inc.) 2$\\eta$m í¬ì„ì‹œí‚¬ìˆ˜ ìˆë‹¤.\n",
    "\n",
    "poolëœ 16S rRNA gene amplicon librariesì™€ phiX controlì€ illumina Incì˜ ë”°ë¼ ìˆ˜í–‰ë˜ê³ , denatured template concentrationì€ 20$\\rho$mì´ë‹¤.\n",
    "\n",
    "Denatured templates are further diluted to 5 $\\rho$M (using Illumina HT1 buffer) and subsequently combined to give an 85% 16S rRNA gene amplicon library and 15% phiX control pool (1000 $\\mu$L total volume). \n",
    "\n",
    "Illumina ë¶„ì„ ì†Œí”„íŠ¸ì›¨ì–´ì˜ ê°œì„ ìœ¼ë¡œ 16S rRNA gene ampliconsì˜ ë” ë§ì€ sequenceë¥¼ ì‚¬ìš©í•  ìˆ˜ ìˆë‹¤.  \n",
    "  \n",
    "MiSeq reagent cartridges are prepared according to the manufacturerâ€™s instructions (Illumina Inc.). The sample pool (1000 $\\mu$L total volume) is loaded into cartridge position 17. \n",
    "Custom 16S rRNA gene Read 1, Index Read, and Read 2 sequencing primers are added directly to cartridge wells containing the standard Illumina Read 1, Index Read, and Read 2 sequencing primers (wells 12, 13, and 14, respectively, 5 $\\rho$L each primer at 100 $\\mu$M concentration (Caporaso et al., 2012)). \n",
    "Primers are added to wells using a long gel loading tip and gently mixed using a plastic Pasteur pipette. \n",
    "Care must be taken to assure that reagents in the cartridge are localized to the bottom of the wells and that no bubbles are present.  \n",
    "The spike-in of PhiX, at least at low levels, is still critical for obtaining usable amplicon data because the optics requires diversity at each nucleotide position, which is not possible with absolutely conserved nucleotides within the 16S rRNA gene (or most other genes of interest). Many users have had difficulty mixing this protocol for custom amplicons with Illuminaâ€™s own indexing protocol, which allows a maximum of 96 samples to be multiplexed per run at the time of writing. \n",
    "It is critical to use either this protocol exactly (allowing arbitrary numbers of custom barcodes) or Illuminaâ€™s barcoding protocol, but not to mix and match steps and reagents. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "![](img/overview.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QIIME workflow overview.\n",
    "The upstream process (ê°ˆìƒ‰ ë°•ìŠ¤)ì´ê³  OTU tableê³¼ phylogenetic treeë¥¼ ìƒì„±í•˜ëŠ” ê²ƒì„ í¬í•¨í•˜ê³  ìˆë©°,  \n",
    "ì‹œí€€ìŠ¤ í”„ë¡œì„¸ì‹± ì½ëŠ”ê²ƒ ì‹œì‘ìœ¼ë¡œ OTU tableê³¼ phylogentic treeë¥¼ ë§Œë“œëŠ”ê²ƒì„ ëë‚œë‹¤.  \n",
    "  \n",
    "The downstream process (íŒŒë€ ë°•ìŠ¤) ë¶„ì„ê³¼ í•´ì„ì´ ì„ì—¬ìˆë‹¤.  \n",
    "ê·¸ë¦¬ê³  OTU tableê³¼ phylogenetic treeë¥¼ ì‹œì‘ìœ¼ë¡œ alpha-, beta-diversity analyses, visualizations, statisticsì„ë¡œ ëë‚œë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. QIIME WORKFLOW FOR CONDUCTING MICROBIAL COMMUNITY ANALYSIS\n",
    "The Illumina MiSeq technologyëŠ” í•œë²ˆì˜ ì‹¤í–‰ì—ì„œ 107ì‹œí€€ìŠ¤ë¥¼ ìƒì„±í•  ìˆ˜ ìˆë‹¤.(Kuczynski et al., 2012).   \n",
    "QIIMEì€ ê¸°ê³„ì—ì„œ ìƒì„±ëœ ìœ ìš©í•œ ì •ë³´(ê°ê°ì˜ cummunity)ì˜ ëŒ€í‘œ ìƒ˜í”Œ ê°€ì ¸ì˜¨ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.1. Upstream analysis Steps  \n",
    "QIIME analysis workflowëŠ” sequencing ì¶œë ¥ë¬¼(fastq files)ê³¼ ìœ ì €ê°€ ìƒì„±í•œ mapping fileìœ¼ë¡œ ì‹œì‘í•œë‹¤. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.1 Demultiplexing and quality-filtering\n",
    "high-throughput sequencingì„ ì‚¬ìš©í•˜ë©´ ì—¬ëŸ¬ ìƒ˜í”Œì„ í•œ sequencing ì‘ì—…ì— ê²°í•© í•  ìˆ˜ ìˆë‹¤. (Kuczynski et al., 2012)  \n",
    "ê·¸ëŸ¬ë‚˜ ê°ê°ì˜ sequenceë¥¼ "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.2 OTU picking\n",
    "ì´ë²ˆ ìŠ¤í…ì€ OTUsì— ì†í•˜ëŠ” clustering preprocesse sequenceì´ë‹¤. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.3 Identify chimeric sequences  \n",
    "PCR amplification processë¥¼ í•˜ëŠ” ë™ì•ˆ, ëª‡ ê°œì˜ amplified sequencesëŠ” multiple prarent sequencesê³¼ chimerasê°€ í¬í•¨ëœ sequencesë¥¼ ìƒì„±í•œë‹¤. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.4 Taxonomy assignment\n",
    "ë‹¤ìŒ ìŠ¤íƒ­ì€ QIIMEì—ì„œ taxonomyë¥¼ ê°ê°ì˜ ëŒ€í‘œì ì¸ sequenceì— ì§€ì •í•˜ëŠ” ê²ƒì´ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.5 Sequence alignment\n",
    "ë‹¤ìŒ ìŠ¤íƒ­ì€ QIIME workflowì´ë©°, sequencesë¥¼ ì •ë ¬í•˜ëŠ” ë‹¨ê³„ì´ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.6 Phylogeny construction\n",
    "ì´ë²ˆ ìŠ¤í…ì€ ì´ì „ë‹¨ê³„ì—ì„œ ìƒì„±ëœ multiple sequenceë¥¼ phylogenetic treeìœ¼ë¡œ ì¶”ë¡ ì„ í•œë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.7 Make OTU table\n",
    "Upstreamì˜ ë§ˆì§€ë§‰ ë¶€ë¶„ì€ OTU table êµ¬ì„±í•˜ëŠ” ê²ƒì´ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.2 Downstream analysis steps  \n",
    "ì¼ë‹¨ OTU tableê³¼ phylogenetic treeì„ ìƒì„±í–ˆìœ¼ë¯€ë¡œ Downstream analysisë¥¼ ì§„í–‰ í•  ìˆ˜ ìˆë‹¤.  \n",
    "ì´ ì‹œì ì—ì„œ ì¶”ì²œí•˜ëŠ” ê²ƒì€ ëŒ€ëŸ‰ì˜ OTUë¥¼ ì´ˆë‹¹(second level)ìœ¼ë¡œ í’ˆì§ˆ í•„í„°ë§ í•˜ëŠ”ê²ƒì´ë‹¤.  \n",
    "ì¦‰, ì´ sequencesì—ì„œ 0.005% ë¯¸ë§Œì˜ OTU sequencesë¥¼ ì œê±°í•˜ëŠ” ê²ƒì´ë‹¤.(see Bokulich et al., 2013 for a detailed description of the effect of this parameter in further downstream analyses)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.1 Taxa summaries\n",
    "ê° sampleì˜ ì²« ë²ˆì§¸ OTU visualizeì€ taxa summaryì´ë‹¤.  \n",
    "taxa summaryëŠ” multiple taxonomic levelsì™€ ê´€ê³„ê°€ ë§ì€ taxa presentì˜ ìƒ˜í”Œ ì§‘í•©ì„ ìš”ì•½í•˜ëŠ” ê²ƒì´ë‹¤.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.2 Diversity analysis\n",
    "ë¯¸ìƒë¬¼ ìƒíƒœí•™(Microbial ecology)ì€ ì„œë¡œ ë‹¤ë¥¸ í™˜ê²½ì—ì„œ ë°•í…Œë¦¬ì•„ êµ°ì§‘ì„ íŠ¹ì„±í™”í•˜ê³  ì´ë ‡ê²Œ ëª¨ì¸ êµ°ì§‘ì—ì„œ ë‹¤ì–‘ì„±ì„ êµ¬í•˜ëŠ” ìš”ì†Œë¥¼ ê²°ì •í•¨ìœ¼ë¡œì¨ ë¯¸ìƒë¬¼ì˜ ë‹¤ì–‘ì„±ì„ ì—°êµ¬í•œë‹¤. ê·¸ë¦¬ê³  alpha-, beta-, gamma-ì˜ ë‹¤ì–‘ì„±ìœ¼ë¡œ ì„œë¡œ ë‹¤ë¥¸ ë‹¤ì–‘ì„±ì„ ì¸¡ì •í•˜ê³  ìœ í˜•ì„ ì •ì˜í•œë‹¤.\n",
    "* Alpha-\n",
    "    * Alpha-diversityëŠ” ë‹¨ì¼ í‘œë³¸ ë˜ëŠ” í™˜ê²½ì—ì„œ ìƒë¬¼ì˜ ë‹¤ì–‘ì„±ì„ ì •ì˜í•œë‹¤.\n",
    "* Beta-\n",
    "    * Beta-diversityëŠ” í‘œë³¸ì´ í™˜ê²½ì— ë”°ë¥¸ ë‹¤ì–‘ì„±ì˜ ì°¨ì´ë¥¼ ë§í•œë‹¤.\n",
    "    * Beta-diversity is the difference in diversities across samples or environments.\n",
    "* Gamma-\n",
    "    * Gamma-diversity(ğ›¾-diversity)ëŠ” ë²”ìœ„ ë˜ëŠ” ë¶€ë¶„ê°™ì´ ê´‘ë²”ìœ„í•œ ì°¨ì›ì—ì„œ ë‹¤ì–‘ì„±ì„ ì¸¡ì •í•œë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://cafe.daum.net/_c21_/bbs_search_read?grpid=n91&fldid=3G4&datanum=102  \n",
    "* Alpha-diversityëŠ” ì¢… í’ë¶€ë„ë¼ëŠ” ê°œë…ì´ ê°€ì¥ ê°€ê¹Œìš´ ì¸¡ì •ë„ ë°©ë²•ì´ê³ , ê°ê¸° ë‹¤ë¥¸ ìƒíƒœê³„ì— ì„œì‹í•˜ëŠ” êµ°ì§‘ ê°„ ì¢…ìˆ˜ë¥¼ ë¹„êµí•  ë•Œ í”íˆ ì‚¬ìš©ëœë‹¤.  \n",
    "  \n",
    "* Beta-diversityëŠ” ì¢… ì¡°ì„±ì´ ë‹¤ì–‘í•œ í™˜ê²½ ê¸°ìš¸ê¸°ì— ë”°ë¼ ì–¼ë§ˆë‚˜ ë³€í•˜ëŠ”ì§€ ê·¸ ì •ë„ë¥¼ ë‚˜íƒ€ë‚´ëŠ” ê²ƒì´ë‹¤.  \n",
    "ì˜ˆë¥¼ ë“¤ì–´  \n",
    "> ì´ë¼ êµ°ì§‘ì´ ê³ ë„ì— ë”°ë¼ ê³„ì† ë³€í•œë‹¤ë©´ ë² íƒ€ ë‹¤ì–‘ì„±ì´ ë†’ë‹¤ê³  ë§ í•  ìˆ˜ ìˆì§€ë§Œ,  \n",
    "ì‚°ë¹„íƒˆì„ ë”°ë¼ ê³„ì† ë™ì¼í•œ ì¢… ì¡°ì„±ì´ ìœ ì§€ëœë‹¤ë©´ ë² íƒ€ ë‹¤ì–‘ì„±ì´ ë‚®ë‹¤ê³  í•  ìˆ˜ ìˆë‹¤.  \n",
    "* Gamma-diversityëŠ” ì§€ë¦¬ì ìœ¼ë¡œ ì¢€ë” ë„“ì€ ë²”ìœ„ì— ì ìš©ë ìˆ˜ ìˆë‹¤.  \n",
    "ë˜‘ê°™ì€ ì„œì‹ì²˜ ê°„ ë˜ëŠ” ì§€ë¦¬ì ì¸ ì§€ì—­ ê°„ ê±°ë¦¬ì— ëŒ€í•œ ì–´ëŠ í•œ ì¢…ì˜ íšŒì „ë¥ (turnover rate)ì„ ì˜ë¯¸í•œë‹¤.    \n",
    "ì¦‰, ê°ë§ˆ ë‹¤ì–‘ì„±ì´ë€ \"ì–´ë–¤ ì¢…ë“¤ì´ ì§€ë¦¬ì ìœ¼ë¡œ ë‹¤ë¥¸ ì§€ì—­ì—ì„œ ê°™ì€ ì¢…ë¥˜ì˜ ì„œì‹ì²˜ë¥¼ ë§Œë‚œ í™•ë¥ \"ì„ ë§í•œë‹¤.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.3 Alpha-diversity analysis\n",
    "QIIMEì€  alpha-diversityì˜ ê²°ê³¼ë¥¼ ë³´ì—¬ì£¼ëŠ” plotì„ ìƒì„±í•˜ì—¬ diversity metricê³¼ different rarefaction levelsì„ ì„ íƒí•˜ë„ë¡ í•  ìˆ˜ ìˆë‹¤.  \n",
    "ì²¨ë¶€ëœ ê·¸ë¦¼(alpha-1)ì€ êµ°ì§‘ ì¤‘ íŠ¹ë³„íˆ ë§ì€ ì¢…ì„ í‰ê°€í•˜ëŠ”ë° ì‚¬ìš©ëœë‹¤.  \n",
    "  \n",
    "  ---\n",
    "![](img/alpha_diversity_1.png)  \n",
    "ê·¸ë¦¼(alpha-1)\n",
    "> ë‹¤ì–‘í•œ OTU pickingì„ í•˜ëŠ” ë°©ë²•ì€ Alpha-diversity curvesì˜ depthì— ë”°ë¼ ê²°ì •ëœë‹¤.  \n",
    " ê°ê°ì˜ ë¼ì¸ì€ alpha-diversityì˜ ë‹¤ì–‘í•œ ê³„í†µì„ QIIMEì˜ PD whole treeì˜ í†µí•´ ë‚˜íƒ€ë‚¸ ê²°ê³¼ì´ë‹¤.\n",
    " (A), (C),and (E)ëŠ” OTU picking protocols(closed-, open-reference, and de nove)ë¡œ ì„ ì •ëœ ë‹¤ë¥¸ ì‹œí€€ìŠ¤ depthì´ë©°, ë‹¤ì–‘í•œ ê°ê°ì˜ ìƒ˜í”Œ í‘œí˜„í•˜ê³  ìˆë‹¤. \n",
    " closed-referenceì—ì„œ OTUì˜ ë°ì´í„° ë² ì´ìŠ¤ë§Œì„ ê³ ë ¤í•˜ì—¬ diversity plateausë¡œ ì‹œí€€ìŠ¤ê°€ ìƒì„±ëœë‹¤.\n",
    " ë˜í•œ closed-referenceëŠ” de novoë¡œ í´ëŸ¬ìŠ¤í„°ë§ ëœ ê²½ìš°ë³´ë‹¤ OTUìˆ˜ ë¥¼ í¬ê²Œ ì¤„ì¸ë‹¤.\n",
    " Alpha-diversity curvesë¥¼ ë¹„êµí•˜ëŠ” ê²ƒì€ ì–´ë µë‹¤. ì™œëƒí•˜ë©´ ì‹œí€€ìŠ¤ depthê°€ ì„œë¡œ ë‹¤ë¥´ë¯€ë¡œ ìƒ˜í”Œ ì¤‘ì—ì„œ ë¹„êµí•˜ëŠ” ê²ƒì€ ì–´ë µê¸° ë•Œë¬¸ì´ë‹¤. \n",
    " (B),(D) ë° (F)ëŠ” ë‘ ì¢…ë¥˜ì˜ ì¥ë¥¼ ì˜ë¯¸í•œë‹¤. ì•¼ìƒí˜•(WT:ì˜¤ëœì§€ìƒ‰), í˜•ì§ˆì „í™˜(TG:ì²­ìƒ‰)\n",
    " * í˜•ì§ˆì „í™˜ë™ë¬¼:\n",
    " ìœ ì „ì ìœ¼ë¡œ ë³€í˜•ëœ ìœ ê¸°ì²´(GMO, Genetically modified organisme), ìœ ì „ì¡°ì‘ ìƒë¬¼"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QIIMEì€ indices(which require a phylogenetic)ì™€ non phylogenetic indices ë‘ê°€ì§€ë¥¼ í¬í•¨í•œ ìˆ˜ì‹­ ê°œì˜ alpha-diversity indicesë¥¼ êµ¬í˜„í–ˆë‹¤.\n",
    "UserëŠ” íŒŒë¼ë¯¸í„° -së¥¼ alpha_diversity.py ìŠ¤í¬ë¦½íŠ¸ì— ì „ë‹¬í•˜ì—¬ alpha-diversity indices(ë³€ìˆ˜) ë¦¬ìŠ¤íŠ¸ë¥¼ ì–»ì„ ìˆ˜ ìˆë‹¤ ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ë¹„êµë¥¼ ìœ„í•´ ë™ì¼í•œ sequence depthë¡œ ë§Œë“¤ì–´ì¤˜ì•¼, Rarefaction curvesë¥¼ í‰ê°€ì— ì‚¬ìš©í•  ìˆ˜ ìˆë‹¤.  \n",
    "ì´ ë°©ì‹ì„ í†µí•˜ì—¬ ìœ„ì˜ ê·¸ë¦¼(alpha-1)ì„ ë„ì¶œ í•˜ì˜€ê³  ë¯¸ìƒë¬¼ êµ°ì§‘ì„ ëŒ€í‘œí•˜ëŠ” ë¹„êµë¥¼ ë½‘ì•„ë‚´ì—ˆë‹¤.  \n",
    "ê·¸ëŸ¬ë‚˜ Sanger sequencingì„ ì‚¬ìš©í• ë•ŒëŠ” rarefaction corvesê°€ ë§ì´ ì‚¬ìš©ë˜ì—ˆì§€ë§Œ,  \n",
    "ìš”ì¦˜ì—ëŠ” ëŒ€ë¶€ë¶„ì˜ êµ°ì§‘ì´ undersample ë˜ì—ˆì„ë•Œ, OTUì˜ ë²”ìœ„ ì™€ ì˜ˆì¸¡ ìˆ˜ ê·¸ë¦¬ê³  ê´€ì¸¡ ìˆ˜ë¥¼ í•˜ë‚˜ì˜ rarefaction depthë¡œ ì‚¬ìš©í•˜ëŠ” ê²ƒì´ ë” ìœ ìš©í•œ ê²½ìš°ê°€ ë§ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.4 Beta-diversity analysis\n",
    "Beta-diversityëŠ” rarefied OTU tableë¡œ ë¶€í„° ê³„ì‚°í•  ìˆ˜ ìˆê³ , ë¯¸ìƒë¬¼ êµ°ì§‘ë“¤ì„ êµ¬ì„±ë¬¼ì„ ê¸°ë°˜í•˜ì—¬ ë¹„êµí•œë‹¤.  \n",
    "alpha-diversityì™€ ê°™ì´ QIIMEì€ ë§ì€ phylogeneticì™€ nonphylogenetic beta-diversity metrics(beta_diversity.py -s ëª…ë ¹ì–´ë¡œ ì‚¬ìš©ì´ ê°€ëŠ¥í•˜ë‹¤.) ê³„ì‚°ì´ ê°€ëŠ¥í•˜ë‹¤.  \n",
    "ì´ì¤‘ì—ì„œ UniFactê°€ ìƒë¬¼í•™ì  ì˜ë¯¸ë¥¼ ì´ë£¨ëŠ” íŒ¨í„´ì„ ì•Œì•„ë‚¼ë•Œ ê°€ì¥ ë³´í¸ì ìœ¼ë¡œ ìœ ìš©í•˜ë‹¤ëŠ” ê²ƒì„ ë°í˜€ëƒˆë‹¤.  \n",
    "> UnifracëŠ” ê°ê°ì˜ êµ°ì§‘ ê³ ìœ  ì§„í™”ëŸ‰ì„ ì¸¡ì •í•œë‹¤.(\n",
    "Unifrac measures the amount of unique evolution within each community with respect to another by calculating the fraction of branch length of the phylogenetic tree that is unique to either one of a pair of communities)  \n",
    "QIIMEì— ë³€í˜•ëœ Unifracì´ êµ¬í˜„ë˜ì–´ ìˆìœ¼ë©°, ê°€ì¤‘ì¹˜ê°€ í¬í•¨ëœ Unifracê³¼ ê°€ì¤‘ì¹˜ê°€ í¬í•¨ë˜ì§€ ì•Šì€ Unifracì´ ìˆë‹¤.  \n",
    "ê°€ì¤‘ì¹˜ê°€ í¬í•¨ëœ Unifrac metricì€ ê° êµ¬ì—­ë§ˆë‹¤ ê° êµ°ì§‘ì˜ OTUì˜ probability massì˜ ì°¨ì´ë¡œ ê°€ì¤‘ì¹˜ê°€ ë¶€ì—¬ë˜ê³ ,  \n",
    "ê°€ì¤‘ì¹˜ê°€ í¬í•¨ë˜ì§€ ì•Šì€ Unifrac metricì€ OTUì˜ ì¡´ì¬/ë¶€ì¬ë§Œ í™•ì¸í•œë‹¤.  \n",
    "* Weighted Unifrac : weighted Unifracì€ taxaê°€ ì¡´ì¬ì—¬ë¶€ ë³´ë‹¤, taxaì˜ ì–‘ì— ì°¨ì´ë¥¼ ì§€ë‹Œ ì„œë¡œ ë‹¤ë¥¸ êµ°ì§‘ íƒì§€í•˜ëŠ”ë° ì¶”ì²œí•œë‹¤.  \n",
    "* Unweighted Unifrac : taxaì˜ ì¡´ì¬ë§Œì„ ê³ ë ¤í•œ ë‹¤ë¥¸ metricsì™€ ë§ˆì°¬ê°€ì§€ë¡œ Unifrac metricëŠ” DNA extraction efficienty, PCR amplification, ë“±ì— ë¯¼ê°í•œ ì„±í–¥ì´ ìˆë‹¤.  \n",
    "  \n",
    "metricsì„ íƒì€ beta-diversity ë¶„ì„ì—ì„œ ì¤‘ìš”í•˜ë©°, ë™ì¼í•œ ë°ì´í„° ì…‹ì—ì„œ êµ°ì§‘ê³¼ gradient patternsíƒì§€í•˜ëŠ” ëŠ¥ë ¥ì´ ë‹¬ë¼ì§„ë‹¤. (Arumugam et al., 2011; Ravel et al., 2012; Schloss &\n",
    "Handelsman, 2006)  \n",
    "See Kuczynski, Liu, et al. (2010) for a detailed discussion of the performance of different nonphylogenetic metrics.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QIIMEì€ ê° ìŒì˜ input sampleì˜ beta-diversities ê³„ì‚°í•˜ì—¬ distance matrixë¥¼ ë§Œë“ ë‹¤.  \n",
    "distance matrixë¥¼ ë§Œë“  ì´í›„ visualizedë¥¼ ì§„í–‰í•˜ë©°, PCoA(Mardia, Kent, & Bibby, 1979)ì™€ hierarchical clustering(Tryon, 1939) ë‘ ê°€ì§€ ëª¨ë‘ visualizedì— ë„ë¦¬ ì‚¬ìš©ë˜ì—ˆë‹¤.  \n",
    "PCoAëŠ” original multidimentsional matrixì˜ maximum amountì˜ ë°ì´í„°ì…‹ì˜ ê´€ì„±ì„ ë¥¼ ìƒˆë¡œìš´ ì§êµì¶•ìœ¼ë¡œ ë³€í™˜í•˜ê³  êµ¬í˜„ë˜ì–´ ìˆëŠ” QIIME ìŠ¤ì¼€ì¼ì— ìˆ˜ì²œê°œì˜ ìƒ˜í”Œë¡œ ë³€í™˜í•œë‹¤.  PCoA transforms the original multidimensional matrix to a new set of orthogonal axes that explain the maximum amount of inertia in the dataset (Gower, 1966; Mardia et al., 1979) and the current implementation in QIIME scales to thousands of samples.  \n",
    "ì§€ê¸ˆì€ ëŒ€ëµì ì¸ ë°©ë²•ìœ¼ë¡œ ìˆ˜ì²œê°œì˜ ìƒ˜í”Œì„ í‰ê°€í•œë‹¤.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![](img/beta-1.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ê·¸ë¦¼(beta-1)\n",
    "> ìœ„ ê·¸ë¦¼ì€ unweighted Unifrac beta-diversity ì´ë‹¤.  \n",
    "* Panels A-Cê¹Œì§€ Datasetì— de novo OTU picking, closed-reference OTU picking, ê·¸ë¦¬ê³  open-reference OTU pickingì„ jackknifed ì¶œë ¥í•œ ê²°ê³¼ì´ë‹¤.  \n",
    "ì„¸ê°€ì§€ ê·¸ë¦¼ì€ OTU picking approacheì— ë”°ë¥¸ ë‹¤ë¥¸ ê²°ê³¼ë¥¼ ë‚˜íƒ€ë‚¸ë‹¤.  \n",
    "ê°ê°ì˜ ì ì€ WT mouse(orange), TG mouse(blue)ë¥¼ ë‚˜íƒ€ë‚¸ë‹¤.  \n",
    "ë‘ê°œì˜ ê·¸ë£¹ì€ ê¹¨ë—í•˜ê²Œ ë¶„ë¦¬ë˜ì§€ ì•Šì•˜ë‹¤.  \n",
    "ê·¸ ì´ìœ ëŠ” ì•„ë§ˆë„ ê¹¨ë—í•˜ê²Œ ì „ì²˜ë¦¬ ë˜ì§€ ì•Šì•˜ê¸° ë•Œë¬¸ì´ë‹¤.(recall ë°©ë²•ì€ ë‹¤ì–‘í•˜ê²Œ ì¡´ì¬í•œë‹¤.)  \n",
    "íƒ€ì›í˜•ì˜ í¬ê¸°ëŠ” jackknife analysisë¡œ ë¶„ì„í•œ ê° ìƒ˜í”Œì˜ ë³€ë™?(ë¶„ì‚°?) ê³„ì‚°ê°’ì„ ë‚˜íƒ€ë‚¸ë‹¤. (The size of the ellipsoids shows the variation for each sample calculated from jackknife analysis)  \n",
    "ìœ„ plotì„ ìƒì„±í•˜ëŠ” ëª…ë ¹ì–´ëŠ” ë‹¤ìŒê³¼ ê°™ë‹¤.  \n",
    "  \n",
    "`jackknifed_beta_diversity.py -i $PWD/denovo_otus/otu_table_filtered.biom -t $PWD/denovo_otus/rep_set.tre -m\n",
    "$PWD/IQ_Bio_16sV4_L001_map.txt -o $PWD/diversity_analysis/jk_denovo\n",
    "-e 7205 -a -O 64`(input parametersëŠ” ë‹¤ë¥¸ OTU picking approachì—ì„œ ì‚¬ìš©ëœ OTU tableì´ ì ìš©ë˜ì•¼í•œë‹¤.)  \n",
    "> * Panel DëŠ” \"keyboard\" data setì—ì„œ ì˜¨ beta-diversity PCoA plotì„ ë‚˜íƒ€ë‚¸ë‹¤.(Fierer et al., 2010)  \n",
    "\"keyboard\" data setì€ ë¯¸ìƒë¬¼ êµ°ì§‘ ìœ ìƒì„±ì„ í†µí•´ ê°œì¸ì˜ í‚¤ë³´ë“œì— ì—°ê²°í•œë‹¤.  \n",
    "ê° ì ì€ 3ëª…ì˜ ê°œì¸ì˜ ì†ëê³¼ í‚¤ë³´ë“œ í‚¤ë¡œ ë¶€í„° ìƒ˜í”Œë§í•œ ë¯¸ìƒë¬¼ êµ°ì§‘ì„ ë‚˜íƒ€ë‚¸ë‹¤.\n",
    "* panel A-Cì™€ ëŒ€ì¡°ì ìœ¼ë¡œ panel DëŠ” PCoA plotì—ì„œ ê° ê°œì²´ì˜ ë¯¸ìƒë¬¼ êµ°ì§‘ì´ ì˜ ë‚˜ëˆ ì§„ê²ƒì„ ë³´ì—¬ì¤€ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QIIMEì€ PCoAë¥¼ KiNG ë·°ì–´ë¥¼ í†µí•´ 3ì°¨ì›ì—ì„œ ì‹œê°í™”í•œë‹¤.  \n",
    "PCoA plotì˜ ì•ˆì •ì„±ì„ í‰ê°€í•˜ê¸°ìœ„í•´ì„œ jackknife resamplingì„ OTU tableì—ì„œ ìˆ˜í–‰ê°€ëŠ¥í•˜ê³ , ê° resamplingëœ tableì— PCoAì ˆì°¨ë¥¼ ì¬ ìˆ˜í–‰í•œë’¤ sample point ì£¼ë³€ì— íƒ€ ì›í˜•ìœ¼ë¡œ ì§‘ê³„ ê²°ê³¼ë¥¼ ploting í•œë‹¤.(ê·¸ë¦¼ beta-1)  \n",
    "UniFracë¿ë§Œ ì•„ë‹ˆë¼ ë§ì€ ë‹¤ì–‘ì„± ì¸¡ì • ë°©ë²•ì´ í‘œë³¸ ë‹¹ ì„œì—´ ìˆ˜ì— ë¯¼ê°í•˜ê²Œ ë°˜ì‘í•˜ê¸° ë•Œë¬¸ì— Jackkiniffeë¥¼ ì¶”ì²œí•œë‹¤.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Jackkiniffeì— ê´€í•œ ë…¼ë¬¸](https://www.researchgate.net/profile/Ruofei_Du/publication/323595926_A_weighted_Jackknife_method_for_clustered_data/links/5b3b9e174585150d23f64029/A-weighted-Jackknife-method-for-clustered-data.pdf?origin=publication_detail)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "![](img/beta-2.png)  \n",
    "ê·¸ë¦¼(beta-2)  \n",
    "> ë‹¤ìŒ ê·¸ë¦¼ì€ Biplotì˜ data setì˜ˆì œì´ë‹¤.  \n",
    "ì´ unweighted Unifrac beta-diversity plotì€ ìœ„ ê·¸ë¦¼ê³¼ ë¹„ìŠ·í•œ í•˜ë‹¤.  \n",
    "5ê°œì˜ ë ˆì´ë¸”ì€ phylum-levelì´ ê°€ì¥ í’ë¶€í•œ 5ê°œì˜ taxaë¥¼ í‘œì‹œí•œê²ƒì´ë‹¤.  \n",
    "ê° ë¶„ë¥˜ ì›ì˜ í¬ê¸°ëŠ” ê° taxon  (The size of the sphere for each taxon is proportional to the mean relative abundance of that taxon across all samples.)  \n",
    "ìœ„ plotì„ ìƒì„±í•˜ëŠ” ëª…ë ¹ì–´ëŠ” ë‹¤ìŒê³¼ ê°™ë‹¤.  \n",
    "  \n",
    "`make_3d_plots.py -i $PWD/diversity_analysis/open_ref/bdiv_even7205/\n",
    "unweighted_unifrac_pc.txt -m $PWD/IQ_Bio_16sV4_L001_map.txt -t $PWD/\n",
    "diversity_analysis/open_ref/taxa_plots/table_mc7205_sorted_L3.txt\n",
    "â€“n_taxa_keep 5 -o $PWD/diversity_analysis/3d_biplot.`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.6 OTU networks  \n",
    "ê°€ë” Network-based analysisëŠ” OUTì˜ ìƒ˜í”Œì´ ì–´ë–»ê²Œ ë¶„í•  ë¬ëŠ”ì§€ ì‹œê°í™” í•˜ëŠ”ê²ƒê³¼ í‘œë³¸ì´ ì„œë¡œ ì–´ë–»ê²Œ ì—°ê´€ ë˜ì–´ ìˆëŠ”ì§€ì— ëŒ€í•´ ì‹œê°í™” í•˜ëŠ”ê²ƒì— ë§¤ìš° ìœ ìš©í•˜ë‹¤.  \n",
    "ê·¸ë ‡ì§€ë§Œ ì´ ë¶„ì„ì€ ë°ì´í„° ì…‹ì´ ëª¨ë‘ í‰ë“±í•˜ê²Œ(equally) ì—°ê²°ë˜ì§€ ì•Šì€ datasetì— ë§Œ ì‚¬ìš©í•˜ëŠ” ê²ƒì´ ì í•©í•˜ë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.7 OTU heatmaps  \n",
    "OTUsì™€ sampleì˜ ê´€ê³„ì— ëŒ€í•œ ì‹œê°í™”ë¥¼ ìœ„í•œ ë‹¤ë¥¸ ë°©ë²•ì€ heatmapì„ ì‚¬ìš©í•˜ëŠ” ê²ƒì´ë‹¤.  \n",
    "heatmapì€ molecular biologyì˜ ë‹¤ë¥¸ ë¶„ì•¼ì—ì„œë„ ë²”ìš©ì ìœ¼ë¡œ ì‚¬ìš©í•œë‹¤."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.8 OTU category significance  \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. OTHER FEATURES  \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. RECOMMENDATIONS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. CONCLUSIONS\n",
    "QIIMEì€  bacterial communityì˜ ë¶„ì„ì— ë§¤ìš° ê°•ë ¥í•œ íˆ´ì´ë‹¤.\n",
    "ì—°êµ¬ì›ë“¤ì´ ë°˜ë³µì ìœ¼ë¡œ í•„ìš”ë¡œ í•˜ëŠ” ì‹œí€€ìŠ¤ í”„ë¡œì„¸ìŠ¤ ì¦‰, raw dataë¥¼ visualizations í•˜ê±°ë‚˜ ê²°ê³¼ë¥¼ í•´ì„í•˜ëŠ”ë° ë§¤ìš° ê°•ë ¥í•˜ë‹¤.\n",
    "ë‘ê°€ì§€ì˜ ì¥ì ì´ QIIMEì„ ë§¤ìš° ìœ ìš©í•˜ê²Œ ë§Œë“ ë‹¤.\n",
    "* ì•Œê³ ë¦¬ì¦˜ì˜ ì¶©ì‹¤ë„ë¡œ ì¸í•´ ë¶„ì„ì´ ì¼ê´€ì ì´ë‹¤.  \n",
    "ì¶©ì‹¤í•˜ê²Œ êµ¬í˜„ëœ ì•Œê³ ë¦¬ì¦˜ìœ¼ë¡œ ì˜¤ë¦¬ì§€ë„ ì €ìê°€ ë§Œë“  ì•Œê³ ë¦¬ì¦˜ì„ ì œëŒ€ë¡œ ë””ìì¸í•˜ê³  ë§Œë“¤ì—ˆê³  í…ŒìŠ¤íŠ¸í•˜ì˜€ê³  ê·¸ë•Œë¬¸ì— QIIMEì€ ë‹¤ë¥¸ platformsì™€ ì°¨ë³„í™” ë˜ê²Œ ë˜ì—ˆë‹¤.   \n",
    "ê·¸ë¦¬ê³  í•œë²ˆ upsteam í”„ë¡œì„¸ìŠ¤ê°€ ëë‚œë‹¤.?   \n",
    "downstream í”„ë¡œì„¸ìŠ¤ëŠ” ì‹œí€€ìŠ¤ platformì˜ same independent ì‚¬ìš©í•œë‹¤.  \n",
    "ìœ„ íŠ¹ì§•ì´ ì§€ì†ì ìœ¼ë¡œ QIIME open sourceê°€ update ë˜ë„ë¡í•˜ì˜€ê³  QIIME usr ì»¤ë®¤ë‹ˆí‹°ê°€ ê¸‰ì„±ì¥í•˜ê²Œ ë˜ëŠ” ê²ƒì„ ì´‰ì§„í•˜ê²Œ ë˜ì—ˆë‹¤.(Caporaso, Kuczynski, et al., 2010)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Downstream ê³¼ Upsteam í”„ë¡œì„¸ìŠ¤ëŠ” ë¶„ì„ì„ ìˆ˜í–‰í•˜ëŠ” ëª‡ê°€ì§€ ì˜µì…˜ì œê³µí•˜ëŠ” í˜•íƒœë¡œ QIIMEì—ê²Œ ì´í–‰í•œë‹¤.  \n",
    "ì´ ë¦¬ë·°ì—ì„œëŠ”, ê° ë‹¨ê³„ì˜ ì›ì¹™ê³¼ ìŠ¤í¬ë¦½íŠ¸ì˜ ê¸°ëŠ¥ ë° ì˜µì…˜ ê°„ ì„ íƒ ë°©ë²•ì— ëŒ€í•´ì„œ ë…¼ì˜í•˜ê³  ì‹œì—°í•œë‹¤.  \n",
    "QIIMEì˜ ì‚¬ìš©ê³¼ ë…ë¦½ì ìœ¼ë¡œ, ì´ ë¦¬ë·°ëŠ” ë˜í•œ high-throughput ì‹œí€€ì‹±ì— ì˜í•´ ìƒì„±ëœ 16S rRNA ì„œì—´ ë¶„ì„ì— ë¯¸ìƒë¶ˆ êµ°ì§‘ ë¶„ì„ì˜ ì „í˜•ì ì¸ ë‹¨ê³„ì˜ ê°œìš”ë¥¼ ì œê³µí•˜ê³  ìˆë‹¤."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
